From d448e45f503084e1baddb430b8bc2e0afdbc04cb Mon Sep 17 00:00:00 2001
From: Martin Kibera <martin.kibera.n@gmail.com>
Date: Tue, 17 Feb 2026 14:44:14 +0300
Subject: [PATCH] skip drift detection on non-chrome

media clock sync using currentTime produces inconsistenly behavior on different browsers
---
 html/static/js/player.js | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/html/static/js/player.js b/html/static/js/player.js
index 820cc9a..4314a52 100644
--- a/html/static/js/player.js
+++ b/html/static/js/player.js
@@ -674,8 +674,12 @@ videoElement.addEventListener("seeking", () => {
 // Drift detection and correction during playback
 let lastBufferCheck = 0; // track last throttled check
 
+// Skip non-chrome
+const isChrome = typeof navigator !== "undefined" && /chrome|chromium/i.test(navigator.userAgent);
+
 videoElement.addEventListener("timeupdate", async () => {
     if (!formatLoader.npa) return;
+    if (!isChrome) return;
     if (freezePlayback) return;
 
     const now = performance.now();
-- 
2.53.0

